readiness-tool/popup.js

```diff
@@ -0,0 +1,231 @@
+  chrome.tabs.executeScript(null, {
+    file: "getPagesSource.js"
+  }, function () {});
+}
+
+/**
+ * Returns all the 3rd party applications found on the website
+ * @param {String} html - String containing all HTML on the page
+ * @return {Object} 
+ */
+function findDetectedApps(html) {
+  let htmlString = html;
+  let linkToApps = (chrome.runtime.getURL('apps.json'));
+  let xhr = new XMLHttpRequest();
+  let detectedApps;
+  let listAllApps;
+
+  xhr.open("GET", linkToApps, true);
+  xhr.onreadystatechange = function () {
+    // Wait until the response is done (onload or onerror).
+    if (xhr.readyState === 4) {
+      listAllApps = JSON.parse(xhr.response).apps;
+      detectedApps = specifySupport(listAllApps, htmlString);
+      showSupportedAppsInView(detectedApps);
+    }
+  }
+  xhr.send();
+  return detectedApps;
+}
+
+/**
+ * Add supported and unsupported applications to the view
+ * @param {Objected} detectedApps - All 3rd Party Applications found on page
+ */
+
+function showSupportedAppsInView(detectedApps) {
+  let supportedAds = document.getElementById("ads-supported");
+  let supportedAnalytics = document.getElementById("analytics-supported");
+  let notSupportedAds = document.getElementById("ads-notSupported");
+  let notSupportedAnalytics = document.getElementById("analytics-notSupported");
+  let notSupportedOther = document.getElementById("other-notSupported");
+  supportedAds.innerHTML = "";
+  supportedAnalytics.innerHTML = "";
+  notSupportedAds.innerHTML = "";
+  notSupportedAnalytics.innerHTML = "";
+  notSupportedOther.innerHTML = "";
+  supportedAds.appendChild(makeList(detectedApps.supported.ads));
+  supportedAnalytics.appendChild(makeList(detectedApps.supported.analytics));
+  notSupportedAds.appendChild(makeList(detectedApps.notSupported.ads));
+  notSupportedAnalytics.appendChild(makeList(detectedApps.notSupported.analytics));
+  notSupportedOther.appendChild(makeList(detectedApps.notSupported.other));
+}
+
+/**
+ * Splits all detected apps into 'supported' and 'not supported'
+ * @param {Object} apps - All possible services that could exist on the page
+ * @param {String} htmlString - String containing all HTML on the page
+ * @return {Object} 
+ */
+function specifySupport(apps, htmlString) {
+  let obj = apps;
+  let foundThis = {
+    "supported": {
+      "ads": [],
+      "analytics": [],
+      "other": []
+    },
+    "notSupported": {
+      "ads": [],
+      "analytics": [],
+      "other": []
+    }
+  };
+
+  //for all the app objects in the apps.JSON file
+  for (let key in obj) {
+    if (obj.hasOwnProperty(key)) {
+      let val = obj[key];
+      //If object has a 'script' key
+      if (val.script != null) {
+        // Sometimes val.script contains an array of regular expressions to check
+        if (typeof (val.script) == 'object') {
+          for (let x in val.script) {
+            let tempScript = val.script[x].split('\\;');
```

Comment:
I couldn't think of a better way to do it. I thought of using a substring but the only issue is that the "\\;" character isn't always there and it would return an empty string in that case.